catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/transformer/mgmp/MgmpTransformer.java

```diff
@@ -0,0 +1,604 @@
+    }
+
+    private void addMetacardSecurityReleasability(final XstreamPathValueTracker pathValueTracker,
+            MetacardImpl metacard, String path, String releasabilityConstant,
+            String disseminationConstants) {
+        String releasability = pathValueTracker.getFirstValue(toPath(path));
+        if (StringUtils.isNotEmpty(releasability)) {
+            addMetacardReleasabilityAndDisseminationControls(releasability,
+                    metacard,
+                    releasabilityConstant,
+                    disseminationConstants);
+        }
+    }
+
+    private void addMetacardReleasabilityAndDisseminationControls(String caveatString,
+            MetacardImpl metacard, String releasability, String disseminationControls) {
+        if (caveatString.startsWith(MgmpConstants.RELEASABLE_TO)) {
+            String[] nations = StringUtils.substringAfter(caveatString,
+                    MgmpConstants.RELEASABLE_TO + " ")
+                    .split("/");
+            List<String> nationsList = Arrays.asList(nations);
+            metacard.setAttribute(releasability, (Serializable) nationsList);
+            metacard.setAttribute(disseminationControls, MgmpConstants.RELEASABLE_TO);
+        } else if (caveatString.endsWith(MgmpConstants.EYES_ONLY)) {
+            String[] nations = StringUtils.substringBefore(caveatString,
+                    " " + MgmpConstants.EYES_ONLY)
+                    .split("/");
+            List<String> nationsList = Arrays.asList(nations);
+            metacard.setAttribute(releasability, (Serializable) nationsList);
+            metacard.setAttribute(disseminationControls, MgmpConstants.EYES_ONLY);
+        } else if (caveatString.endsWith(MgmpConstants.EYES_DISCRETION)) {
+            String[] nations = StringUtils.substringBefore(caveatString,
+                    " " + MgmpConstants.EYES_DISCRETION)
+                    .split("/");
+            List<String> nationsList = Arrays.asList(nations);
+            metacard.setAttribute(releasability, (Serializable) nationsList);
+            metacard.setAttribute(disseminationControls, MgmpConstants.EYES_DISCRETION);
+        }
+    }
+
+    private void addMetacardResourceSecurityClassification(
+            final XstreamPathValueTracker pathValueTracker, MetacardImpl metacard) {
+        addFieldToMetacardIfPresent(pathValueTracker,
+                metacard,
+                MgmpConstants.RESOURCE_SECURITY_PATH,
+                Security.RESOURCE_CLASSIFICATION);
+    }
+
+    private void addMetacardMetadataSecurityClassification(
+            final XstreamPathValueTracker pathValueTracker, MetacardImpl metacard) {
+        addFieldToMetacardIfPresent(pathValueTracker,
+                metacard,
+                MgmpConstants.METADATA_SECURITY_PATH,
+                Security.METADATA_CLASSIFICATION);
+    }
+
+    private void addMetacardDataQuality(MetacardImpl metacard) {
+
+        try (InputStream inputStream = getSourceInputStream()) {
```

Comment:
This may have to be a future enhancement, but re-processing the input stream like this isn't ideal.  We may need some smarts in the DDF [XstreamTreeWriter](https://github.com/codice/ddf/blob/master/catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/XstreamTreeWriter.java) to handle complex elements like data quality.